pxf/pxf-json/src/test/java/org/apache/hawq/pxf/plugins/json/JsonExtensionTest.java

```diff
@@ -0,0 +1,288 @@
+import org.apache.hawq.pxf.api.Fragmenter;
+import org.apache.hawq.pxf.api.ReadAccessor;
+import org.apache.hawq.pxf.api.ReadResolver;
+import org.apache.hawq.pxf.api.io.DataType;
+import org.apache.hawq.pxf.plugins.hdfs.HdfsDataFragmenter;
+import org.apache.hawq.pxf.plugins.json.JsonAccessor;
+import org.apache.hawq.pxf.plugins.json.JsonResolver;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class JsonExtensionTest extends PxfUnit {
+
+	private List<Pair<String, DataType>> columnDefs = null;
+	private List<Pair<String, String>> extraParams = new ArrayList<Pair<String, String>>();
+
+	@Before
+	public void before() {
+
+		columnDefs = new ArrayList<Pair<String, DataType>>();
+
+		columnDefs.add(new Pair<String, DataType>("created_at", DataType.TEXT));
+		columnDefs.add(new Pair<String, DataType>("id", DataType.BIGINT));
+		columnDefs.add(new Pair<String, DataType>("text", DataType.TEXT));
+		columnDefs.add(new Pair<String, DataType>("user.screen_name", DataType.TEXT));
+		columnDefs.add(new Pair<String, DataType>("entities.hashtags[0]", DataType.TEXT));
+		columnDefs.add(new Pair<String, DataType>("coordinates.coordinates[0]", DataType.FLOAT8));
+		columnDefs.add(new Pair<String, DataType>("coordinates.coordinates[1]", DataType.FLOAT8));
+	}
+
+	@After
+	public void cleanup() throws Exception {
+		columnDefs.clear();
+		extraParams.clear();
+	}
+
+	@Test
+	public void testCompressedMultilineJsonFile() throws Exception {
+
+		extraParams.add(new Pair<String, String>("IDENTIFIER", "created_at"));
+
+		List<String> output = new ArrayList<String>();
+
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547115253761,text1,SpreadButter,tweetCongress,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547123646465,text2,patronusdeadly,,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547136233472,text3,NoSecrets_Vagas,,,");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/tweets.tar.gz"), output);
+	}
+
+	@Test
+	public void testMaxRecordLength() throws Exception {
+
+		// variable-size-objects.json contains 3 json objects but only 2 of them fit in the 27 byte length limitation
+
+		extraParams.add(new Pair<String, String>("IDENTIFIER", "key666"));
+		extraParams.add(new Pair<String, String>("MAXLENGTH", "27"));
+
+		columnDefs.clear();
+		columnDefs.add(new Pair<String, DataType>("key666", DataType.TEXT));
+
+		List<String> output = new ArrayList<String>();
+
+		output.add("small object1");
+		// skip the large object2 XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+		output.add("small object3");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/variable-size-objects.json"), output);
+	}
+
+	@Test
+	public void testDataTypes() throws Exception {
+
+		// TDOO: The BYTEA type is not tested!!! Current oneField.val = val.asText().getBytes(); convention is
```

Comment:
Sure. let's live it as it is for now. Just as a known issue and address it in a separate (JIRA?) task? 
I don't know what should be the desired value but array ref. doesn't seems right to me. Perhaps @adamjshook  can help.